### PR TITLE
fix(sampler): default temperature to 1.0 and torch.zeros to avoid NaN logits

### PR DESCRIPTION
**Pull request by rebel-eunji**
### 🚀 Summary of Changes

Currently, top_k_top_p produces out of vocab index when logits is nan.
To resolve it, we prevented nan logits

1. default temperature to 1.0
     a. padded logits are assigned temperature=0.0.
     b. This produces sampled token ids outside the vocab range, which then raises an out-of-bounds error in the torch.compile'd torch.gather op on CPU.
     c. So this PR defaults temperature to 1.0.
3. Initialize logits with `torch.zeros`

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] 🚀 Release (`release`)
* [ ] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [x] 🛠 Bug fix (`fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
5. Verify output: `...`
6. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [ ] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

**Review by rebel-jonghewk — APPROVED** [review]
LGTM. It would be great to add a unit test as well.
